app/models/odk/response_pattern_parser.rb

```diff
@@ -18,9 +18,14 @@ def join_tokens(tokens)
     end
 
     # Returns the output fragment for the given target questioning.
-    def build_output(other_qing)
-      xpath = src_item.xpath_to(target_qing_or_subqing(other_qing))
-      other_qing.has_options? ? itext_expr(xpath) : xpath
+    def build_output(other_qing, option_value: false)
+      xpath = if option_value
+                other_qing.xpath_to_option_value
+              else
+                src_item.xpath_to(target_qing_or_subqing(other_qing))
+              end
+
+      other_qing.has_options? && !option_value ? itext_expr(xpath) : xpath
```

Comment:
Since we just return `xpath` directly if option_value, let's simplify this and just do it earlier:

```ruby
return other_qing.xpath_to_option_value if option_value
# ...code that was there before
```

Reply:
done!